course6/toyapp_alexs/displaylib/matrix.h

```diff
@@ -1,37 +1,38 @@
 #pragma once
 
-#include "vector3d.h"
 #include <array>
 
 namespace Display {
 
-template<std::size_t A, std::size_t B>
+template<std::size_t A, std::size_t B, class T = double>
 class Matrix
 {
 public:
 	Matrix() {}
-	Matrix(std::array<std::array<double, B>, A> data): mMatrix(data) {}
 
-	std::array<double, B>& operator [](int i)
+	template<class... E>
+	Matrix(E... data): mMatrix {static_cast<T>(data)...} {}
+
+	std::array<T, B>& operator [](int i)
```

Comment:
Does it return row or column? Is operator[] good API for returning row or column?